### PR TITLE
Add `thread` attribute to `Message` class

### PR DESCRIPTION
**Pull request by Ratery**
## Summary
Added `thread` attribute to the `Message` class. It returns the thread which was created from the message.
Also then `message.create_thread` is used `message.thread` will be automatically updated.

## Checklist
- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by plun1331** [comment]
I would make the updating of the thread match the various `_handle_x` methods

File: discord/message.py [hunk 1]
```diff
@@ -536,6 +536,9 @@ class Message(Hashable):
     channel: Union[:class:`TextChannel`, :class:`Thread`, :class:`DMChannel`, :class:`GroupChannel`, :class:`PartialMessageable`]
         The :class:`TextChannel` or :class:`Thread` that the message was sent from.
         Could be a :class:`DMChannel` or :class:`GroupChannel` if it's a private message.
+    thread: Optional[:class:`Thread`]
+        The thread created from a message, if any.
```
Comment (Lulalaby) [inline discord/message.py:540]:
```suggestion
        The thread created from this message, if any.
```

File: discord/message.py [hunk 2]
```diff
@@ -721,6 +725,12 @@ def __init__(
         except KeyError:
             self.interaction = None
 
+        self.thread: Optional[Thread]
+        try:
+            self.thread = Thread(guild=self.guild, state=state, data=data['thread'])
```
Comment (Lulalaby) [inline discord/message.py:730]: Not sure about this..
Too many API calls.

**Review by Lulalaby — CHANGES_REQUESTED** [review]

**Comment by Lulalaby** [comment]
**Reminder:**

⚠️ We currently have a feature freeze

File: discord/message.py (hunk 2 above)
Comment (Ratery) [reply discord/message.py:730]: I could be mistaken but I don't think it's an API call. Just see discord.Thread source code.

**Comment by Ratery** [comment]
> **Reminder:**
> 
> ⚠️ We currently have a feature freeze

Yes, I know. But this PR might be merged before 2.1 version release.

**Review by Dorukyum — DISMISSED** [review]

**Comment by Dorukyum** [comment]
I think we can include this in 2.0 since threads have been added in 2.0 too.

**Comment by krittick** [comment]
Please resolve conflicts. Thanks!

**Comment by Dorukyum** [comment]
Resolved merge conflicts.

File: discord/message.py [hunk 3]
```diff
@@ -580,6 +580,9 @@ class Message(Hashable):
     channel: Union[:class:`TextChannel`, :class:`Thread`, :class:`DMChannel`, :class:`GroupChannel`, :class:`PartialMessageable`]
         The :class:`TextChannel` or :class:`Thread` that the message was sent from.
         Could be a :class:`DMChannel` or :class:`GroupChannel` if it's a private message.
+    thread: Optional[:class:`Thread`]
+        The thread created from this message, if any.
+        .. versionadded:: 2.0
```
Comment (Dorukyum) [inline discord/message.py:585]: I think this is the correct formatting here
```suggestion

        .. versionadded:: 2.0
```

**Comment by plun1331** [comment]
Should this update a cached message's thread when a thread create/update event is received?

**Comment by Lulalaby** [comment]
> Should this update a cached message's thread when a thread create/update event is received?

I would say yes

**Pull request closed without merging**